### DISCUSSION TITLE
Cannot get component libraries to work with the suggested React + Vite

### DISCUSSION THREAD

**Discussion by vim-diesel** [tailwindlabs/tailwindcss#12378] [Help] [answered]
The Tailwind docs suggest not to use create-react-app, but when using vite, none of the component libraries work. For example, in the tailwind component library, it says to use the following config:

```
<!--
  This example requires some changes to your config:

  // tailwind.config.js
  module.exports = {
    // ...
    plugins: [
      // ...
      require('@tailwindcss/forms'),
    ],
  }
-->
```
[Tailwind UI Component](https://tailwindui.com/components/application-ui/forms/form-layouts)
But with vite, this is broken. neither module.exports or require works anymore. 

It is the same with every other component library, they have the same documentation. I see no way to make the libraries work with Vite, unless there is some other keyword than 'require' I am supposed to know,

**Comment by wongjn** [comment]
How does your project break? Is there any error messages you can provide? Or otherwise, how does it behave vs how you expect it to behave?

**Reply by vim-diesel** [reply]
After creating a project using the Tailwind guide (https://tailwindcss.com/docs/guides/vite) any attempt to use any Tailwind plugins as described in the readme (https://github.com/tailwindlabs/tailwindcss-forms) results in an error:

![image](https://github.com/tailwindlabs/tailwindcss/assets/37084860/26de8134-1731-41f0-8255-f0b938f2fa1b)

same goes for any of the popular libraries I have tried to use (preline, tailwind elements...)

**Reply by wongjn** [reply, marked as answer]
You seem to be mixing JavaScript module syntaxes. `export default` is ESM syntax while `require()` is CommonJS. Consider picking the most suitable syntax for your project.

For ESM:

```js
import foo from 'preline/plugin';

export default {
  // …
  plugins: [foo],
};
```

For CommonJS:

```js
module.exports = {
  // …
  plugins: [require('preline/plugin')],
};
```

**Comment by vim-diesel** [comment]
I may have found a fix, although I'm not sure if it is a fix or a bandaid:

To specify environments in a configuration file, use the env key and specify which environments you want to enable by setting each to true. For example, the following enables the browser, es6 and Node.js environments:

In your .eslintrc.js file ;
```
env: {
   browser: true,
   node: true,    <<<<--- Add this
   es6: true
 },
```
